### PR TITLE
pet dehydrated fish to make him nice to you

### LINKED ISSUE #13870: cannot pet dehydrated fish
## Description
in /tg/ petting fish then hydrating makes him like you

**Reproduction**
1. hold fish
2. press z
3. nothing

**Additional context**
https://tgstation13.org/wiki/Syndicate_Items#Dehydrated_Space_Carp

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
fixes #13870

you can pet a pettable friend (currently dehydrated fishe) and you will be friends
once something with friends is hydrated it cannot be petted anymore!!!
fishe bought from uplink will not start prepetted, nukie teams still need to all pet it.

**Media**

urist petted bottom fish, but not top fish...

https://user-images.githubusercontent.com/39013340/225717585-c2b08ea7-a225-40ac-9714-26f1dc88b222.mp4

- [X] I have added a video to this PR showcasing its changes ingame

**Changelog**
:cl:
- tweak: Dehydrated space carp can now be petted to make them nice to you. Have the whole station pet him and be friends forever!

### CODE REVIEW

File: Content.Server/Friends/Systems/FriendsSystem.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            return;
+        }
+
+        // you have made a new friend :)
+        _popups.PopupEntity(Loc.GetString(comp.SuccessString, ("target", uid)), user, user);
+        friends.Friends.Add(user);
+        args.Handled = true;
+    }
+
+    protected void OnRehydrated(EntityUid uid, FriendsComponent comp, ref GotRehydratedEvent args)
```
Comment (EmoGarbage404) [inline Content.Server/Friends/Systems/FriendsSystem.cs:38]: test fail on this event

File: Content.Server/Friends/Systems/FriendsSystem.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
+using Content.Server.Chemistry.Components;
+using Content.Server.Friends.Components;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Popups;
+
+namespace Content.Server.Friends.Systems;
+
+public sealed class FriendsSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPopupSystem _popups = default!;
```
Comment (EmoGarbage404) [inline Content.Server/Friends/Systems/FriendsSystem.cs:10]: probably should be just `_popup` (without the s)

**Review by EmoGarbage404 — COMMENTED** [review]
mostly looks fine. Maybe consider renaming "friends" to something like "factionException" and moving all the namespaces to NPC code. I'll have to get an opinion from @metalgearsloth

**Comment by deltanedas** [comment]
i have methods `IgnoreEntity`, `IgnoreEntities` and `IsIgnored`, what should it be called then? still `FactionExceptionSystem`? im bad with names h

**Comment by deltanedas** [comment]
@EmoGarbage404 tested ingame still works, should i change name of it again or is it ok

**Comment by Elijahrane** [comment]
> adds the friends system, npcs wont attack anyone if they are friends

this will add duplicated behavior if both this and https://github.com/space-wizards/space-station-14/pull/13557 are merged

**Comment by deltanedas** [comment]
i guess can change it to depend on that too, what would the equivalent of `FactionExceptionSystem` be here?

**Comment by Elijahrane** [comment]
> i guess can change it to depend on that too, what would the equivalent of `FactionExceptionSystem` be here?

add the uid of an entity to FactionComponent.ExceptionalFriendlies and it will get pruned whenever it checks for nearby hostile entities

**Comment by deltanedas** [comment]
with your faction changes fish will start prepetted when bought very epic

**Comment by deltanedas** [comment]
@Elijahrane due to access i cannot just `ExceptionalFriendlies.Contains(uid)`, what should i do

**Comment by Elijahrane** [comment]
> @Elijahrane due to access i cannot just `ExceptionalFriendlies.Contains(uid)`, what should i do

use the public api for adding friendlies

**Comment by deltanedas** [comment]
> you shouldn't need it for anything in this PR?

to test if the user is already an exception when petting to change the text to say fishe likes you now

**Comment by deltanedas** [comment]
it needs execute access since technically Contains is a function, should probably just add a function like how i had IsIgnored

**Comment by deltanedas** [comment]
still works fine with latest factionstuff, can be merged after dependencies if no other issues are found

**Comment by deltanedas** [comment]
@EmoGarbage404 no longer using ranes thing anything to change or good to merge

**Review by DrSmugleaf — APPROVED** [review]

**Pull request merged**